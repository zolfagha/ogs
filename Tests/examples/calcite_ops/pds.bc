GeoSys-BC: Boundary Conditions ------------------------------------------------
#BOUNDARY_CONDITION
 $PCS_TYPE
  GROUNDWATER_FLOW
 $PRIMARY_VARIABLE 
  HEAD
 $GEO_TYPE
  POINT POINT0
 $DIS_TYPE
  CONSTANT 0.13229
  
#BOUNDARY_CONDITION
 $PCS_TYPE
  GROUNDWATER_FLOW
 $PRIMARY_VARIABLE 
  HEAD
 $GEO_TYPE
  POINT POINT1
 $DIS_TYPE
  CONSTANT 0.0

#BOUNDARY_CONDITION
 $PCS_TYPE
  REACT_TRANS_OPS
 $PRIMARY_VARIABLE 
  H+
 $GEO_TYPE
  POINT POINT0
 $DIS_TYPE
  CONSTANT  1.e-07

#BOUNDARY_CONDITION
 $PCS_TYPE
  REACT_TRANS_OPS
 $PRIMARY_VARIABLE 
  Ca2+
 $GEO_TYPE
  POINT POINT0
 $DIS_TYPE
  CONSTANT  1.0000e-012

#BOUNDARY_CONDITION
 $PCS_TYPE
  REACT_TRANS_OPS
 $PRIMARY_VARIABLE 
  Mg2+
 $GEO_TYPE
  POINT POINT0
 $DIS_TYPE
  CONSTANT 1.0e-03  

#BOUNDARY_CONDITION
 $PCS_TYPE
  REACT_TRANS_OPS
 $PRIMARY_VARIABLE 
  Cl-
 $GEO_TYPE
  POINT POINT0
 $DIS_TYPE
  CONSTANT 2.0e-03  

#BOUNDARY_CONDITION
 $PCS_TYPE
  REACT_TRANS_OPS
 $PRIMARY_VARIABLE 
  CO32-
 $GEO_TYPE
  POINT POINT0
 $DIS_TYPE
  CONSTANT  4.6738e-012

#BOUNDARY_CONDITION
 $PCS_TYPE
  REACT_TRANS_OPS
 $PRIMARY_VARIABLE 
  HCO3-
 $GEO_TYPE
  POINT POINT0
 $DIS_TYPE
  CONSTANT  9.9954e-011

#BOUNDARY_CONDITION
 $PCS_TYPE
  REACT_TRANS_OPS
 $PRIMARY_VARIABLE 
  Calcite
 $GEO_TYPE
  POINT POINT0
 $DIS_TYPE
  CONSTANT 3.9516e-12 

#BOUNDARY_CONDITION
 $PCS_TYPE
  REACT_TRANS_OPS
 $PRIMARY_VARIABLE 
  Dolomite(dis) 
 $GEO_TYPE
  POINT POINT0
 $DIS_TYPE
  CONSTANT 1.1755e-12

#STOP

