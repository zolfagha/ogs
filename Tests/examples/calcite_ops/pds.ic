GeoSys-IC  Initial Conditions ------------------------------------------------
#INITIAL_CONDITION
 $PCS_TYPE
  GROUNDWATER_FLOW
 $PRIMARY_VARIABLE 
  HEAD
 $GEO_TYPE
  DOMAIN
 $DIS_TYPE
  CONSTANT 1.0

#INITIAL_CONDITION
 $PCS_TYPE
  REACT_TRANS_OPS
 $PRIMARY_VARIABLE 
  H+
 $GEO_TYPE
  DOMAIN
 $DIS_TYPE
  CONSTANT  1.99e-8

#INITIAL_CONDITION
 $PCS_TYPE
  REACT_TRANS_OPS
 $PRIMARY_VARIABLE 
  Ca2+
 $GEO_TYPE
  DOMAIN
 $DIS_TYPE
  CONSTANT  1.23e-2

#INITIAL_CONDITION
 $PCS_TYPE
  REACT_TRANS_OPS
 $PRIMARY_VARIABLE 
  Mg2+
 $GEO_TYPE
  DOMAIN
 $DIS_TYPE
  CONSTANT 1.0e-10  //0.0

#INITIAL_CONDITION
 $PCS_TYPE
  REACT_TRANS_OPS
 $PRIMARY_VARIABLE 
  Cl-
 $GEO_TYPE
  DOMAIN
 $DIS_TYPE
  CONSTANT 2.0e-10  //0.0

#INITIAL_CONDITION
 $PCS_TYPE
  REACT_TRANS_OPS
 $PRIMARY_VARIABLE 
  CO32-
 $GEO_TYPE
  DOMAIN
 $DIS_TYPE
  CONSTANT  2.8847e-7

#INITIAL_CONDITION
 $PCS_TYPE
  REACT_TRANS_OPS
 $PRIMARY_VARIABLE 
  HCO3-
 $GEO_TYPE
  DOMAIN
 $DIS_TYPE
  CONSTANT  1.23e-4

#INITIAL_CONDITION
 $PCS_TYPE
  REACT_TRANS_OPS
 $PRIMARY_VARIABLE 
  Calcite
 $GEO_TYPE
  DOMAIN
 $DIS_TYPE
  CONSTANT 2.07e-4  //about the value from Gems (in mol/kg water)

#INITIAL_CONDITION
 $PCS_TYPE
  REACT_TRANS_OPS
 $PRIMARY_VARIABLE 
  Dolomite(dis) 
 $GEO_TYPE
  DOMAIN
 $DIS_TYPE
  CONSTANT 9.0949e-029

#STOP

