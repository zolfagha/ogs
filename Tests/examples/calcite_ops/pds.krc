GeoSys Reactions
#KINREACTIONDATA      
 $SOLVER_TYPE
	1                ; Solver type
 $RELATIVE_ERROR	
	1.e-6		 ; rel. Error
 $MIN_TIMESTEP	
	1.e-6		; min Timestep
 $INITIAL_TIMESTEP	
	1.e-4       ; init Timestep
 $BACTERIACAPACITY
  100.0
 $NO_REACTIONS
   POLYLINE PLY_BC_LEFT

#REACTION ; 
 $NAME
   R1 ;
 $TYPE
   MOB_EQ_REACT;
 $EQUATION
   1 H+ + 1 CO32- = 1 HCO3- ; 
 $EQ_CONSTANT
 2.132615e+10           ; K values

#REACTION ; 
 $NAME
   R2 ;
 $TYPE
   MIN_EQ_REACT;
 $EQUATION
   1 CO32- + 1 Ca2+ = 1 Calcite;
 $EQ_CONSTANT
 3.0192826e+08           ; K values

#REACTION ; 
 $NAME
   R3 ;
 $TYPE
   MIN_EQ_REACT;
 $EQUATION
   2 CO32- + 1 Ca2+ + 1 Mg2+ = 1 Dolomite(dis); 
 $EQ_CONSTANT
 1.229679319e+17          ; K values
   
#STOP
