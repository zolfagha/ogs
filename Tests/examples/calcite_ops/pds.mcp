#COMPONENT_PROPERTIES  ; comp0
 $NAME
  H+ ;
 $MOBILE
  1; MOBILE-Flag: 0=imMOBILEe, 1=MOBILEe/transported
 $DIFFUSION
  1  0.0e-9             	; diffusion model type, diffusion constant

#COMPONENT_PROPERTIES  ; comp1
 $NAME
  Ca2+ ;
 $MOBILE
  1; MOBILE-Flag: 0=imMOBILEe, 1=MOBILEe/transported
 $DIFFUSION
  1  0.0e-9             	; diffusion model type, diffusion constant

#COMPONENT_PROPERTIES  ; comp2
  $NAME
  Mg2+ ;
 $MOBILE
  1; MOBILE-Flag: 0=imMOBILEe, 1=MOBILEe/transported
 $DIFFUSION
  1  0.0e-9             	; diffusion model type, diffusion constant

#COMPONENT_PROPERTIES  ; comp3
 $NAME  ; 
  Cl- ; 
 $MOBILE
  1; MOBILE-Flag: 0=imMOBILEe, 1=MOBILEe/transported
 $DIFFUSION
  1  0.0e-9             	; diffusion model type, diffusion constant

#COMPONENT_PROPERTIES  ; comp4
 $NAME  ; 
  CO32- ; 
 $MOBILE
  1; MOBILE-Flag: 0=imMOBILEe, 1=MOBILEe/transported
 $DIFFUSION
  1  0.0e-9             	; diffusion model type, diffusion constant

#COMPONENT_PROPERTIES  ; comp5
 $NAME  ; 
  HCO3- ; 
 $MOBILE
  1; MOBILE-Flag: 0=imMOBILEe, 1=MOBILEe/transported
 $DIFFUSION
  1  0.0e-9             	; diffusion model type, diffusion constant
  
#COMPONENT_PROPERTIES  ; comp6
 $NAME
  Calcite ;
 $MOBILE
  0; MOBILE-Flag: 0=imMOBILEe, 1=MOBILEe/transported
 $DIFFUSION
  0             	; diffusion model type, diffusion constant
#COMPONENT_PROPERTIES  ; comp7
 $NAME
  Dolomite(dis) ;
 $MOBILE
  0; MOBILE-Flag: 0=imMOBILEe, 1=MOBILEe/transported
 $DIFFUSION
  0             	; diffusion model type, diffusion constant
#STOP
