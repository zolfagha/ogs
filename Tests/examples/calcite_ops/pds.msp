GeoSys-MSP: Material Solid Properties ------------------------------------
#SOLID_PROPERTIES
  $DENSITY
    1 1.80000e+003
#STOP


;-------------------------------------------------------

#SOIL_PROPERTIES
 1  1.000000e+000             ; dimension, (area), 
 0                            ; porosity model
 0.32  1.000000e+000          ; porosity, tortuosity
 0                            ; mobile immobile model
 0                            ; lithological component
 0                            ; maximum sorption model
 0 ; nonlinear flow parameter
 0.000000e+000 ; storativity
 0 0  3.000000e-07 ; permeability model, permeability tensor, permeabilities
 0  0.000000e+000  0.000000e+000  0.000000e+000  0.000000e+000  0.000000e+000  0.000000e+000  0.000000e+000 ; k-S function
 0.000000e+000  0.000000e+000  0.000000e+000  0.000000e+000  0.000000e+000  0.000000e+000  0.000000e+000 ; p-S function
; 0.100000e+000  0.000000e+000 ; mass dispersion parameters
 0.00670000e+000  0.000000e+000 ; mass dispersion parameters
 0.000000e+000  0.000000e+000 ; heat dispersion parameters
 1.800000e+003  0.000000e+000 ; rock density, heat capacity
 0  0.000000e+000 ; heat conductivity parameters

