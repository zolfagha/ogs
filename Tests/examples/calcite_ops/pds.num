GeoSys-NUM: Numerical Parameter ----------------------------------------
#NUMERICS
 $PCS_TYPE
  GROUNDWATER_FLOW
 $ELE_GAUSS_POINTS
  3
 $LINEAR_SOLVER 
; method error_tolerance max_iterations theta precond storage
  2      6 1.e-014       1000           1.0   1       2
#NUMERICS
 $PCS_TYPE
  REACT_TRANS_OPS
 $LINEAR_SOLVER 
; method error_tolerance max_iterations theta precond storage
  2      6 1.e-014       1000           0.5   1       2
 $ELE_GAUSS_POINTS
  3
#STOP

; 2 Numerical Method ---------------------------------------------------
#NUMERICS
 0 ; numerical method
 PRESSURE ; name of unknown variable
 2 ; gauss points
 1.000000 ; time collocation
 0.000000 ; upwind parameter
 0  0.000000 ; lagrange methode, quality parameter
#NUMERICS
 1 ; numerical method
 TRANSPORT ; name of unknown variable
 2 ; gauss points
 1.000000 ; time collocation
 0.000000 ; upwind parameter
 0  0.000000 ; lagrange methode, quality parameter

#NUMERICS
 1     ; method
TRANSPORT_SORP ; name
 2     ; gaussian points
 1.0   ; time collocation
 0.0   ; upwind parameter
 0 0.0 ; lagrange


; 4.1 Linear Solver Properties -----------------------------------------
#LINEAR_SOLVER_PROPERTIES_PRESSURE
 2  0  1 ; method, norm, preconditioning
 1000  0 ; maximum iterations, repeating
 6  1.000000e-014 ; criterium, absolute error
 0 ; kind
 2 ; matrix storage technique
#LINEAR_SOLVER_PROPERTIES_CONCENTRATION
 2  0  1 ; method, norm, preconditioning
 1000  0 ; maximum iterations, repeating
 6  1.000000e-014 ; criterium, absolute error
 0 ; kind
 2 ; matrix storage technique
