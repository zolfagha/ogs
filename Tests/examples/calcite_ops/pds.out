GeoSys-OUT: Output ------------------------------------------------
#OUTPUT // profile
 $NOD_VALUES
  HEAD
  H+
  Ca2+
  Mg2+
  CO32-
  Cl-
  HCO3-
  Calcite
  Dolomite(dis)
 $GEO_TYPE
  DOMAIN
 $DAT_TYPE
  PVD
 $TIM_TYPE
  STEPS 10
#STOP

